### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by rachidelaid**
> tasks done:
- created `Tasks` class inside `class.js` with the list and all the needed methods inside it.
- added an event listener on the check box to update the `completed` property.
- added `dragging.js` with the logic for dragging and dropping tasks.

### CODE REVIEW

File: src/class.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  sort(oldIndex, newIndex) {
+    this.list[oldIndex - 1].index = newIndex;
+    localStorage.setItem('tasks', JSON.stringify(this.list));
+  }
+
+  clearCompleted() {
+    this.list = this.list.sort((a, b) => a.index - b.index);
+    this.list = this.list.filter((t) => !t.completed);
```
Comment (omar-labana) [inline src/class.js:38]: - [ ] _[OPTIONAL]_ Filter the array before sorting if u wanna reset the indexes. this should take one less iteration per deleted task.
- [ ] The `clear all completed` button isn't working (no console errors and the localstorage isn't updated).

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @rachidelaid, 

Excellent work so far! 👌
You have implemented all the required features correctly with no errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/class.js (hunk 1 above)
Comment (rachidelaid) [reply src/class.js:38]: the `clear all completed` is working for me. make sure you mark some tasks as complete first.

https://user-images.githubusercontent.com/42470159/149208428-a6c70e91-3664-4881-8ec5-86821c81ca85.mp4

**Review by HENRYKC24 — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: 
Approved
## Hi @rachidelaid :wave:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**